hardware/src/lane/vmfpu.sv

```
@@ -617,22 +687,51 @@ module vmfpu import ara_pkg::*; import rvv_pkg::*; import fpnew_pkg::*; #(
         if ((vinsn_issue_mul && vmul_in_ready) || (vinsn_issue_div && vdiv_in_ready) || (vinsn_issue_fpu && vfpu_in_ready)) begin
           // Acknowledge the operands of this instruction
           mfpu_operand_ready_o = operands_ready;
-          // Acknowledge the mask unit
-          mask_ready_o         = ~vinsn_issue_q.vm;
 
+          // Update the element issue counter and the related issue_be signal for the divider
           begin
             // How many elements are we issuing?
             automatic logic [3:0] issue_element_cnt = (1 << (int'(EW64) - int'(vinsn_issue_q.vtype.vsew)));
+            automatic logic [3:0] issue_element_cnt_narrow = (1 << (int'(EW64) - int'(vinsn_issue_q.vtype.vsew))) / 2;
```

Is this handling of narrowing instructions aligned with the VALU?

---

The main differences should be:
 - The FPU shuffles the result outside the FPU, while the ALU does it inside the ALU module.
 - I need to tell the FPU which elements are active to suppress the exceptions from the inactive elements.
 - Differences in coding style, also because the FPU needs two decoupled processes for the issue and the commit.

---

Sure. Please PnR the updated Ara lane, with conversions, and let us check whether this implementation does not lead to a timing degradation. If that is the case, we can resolve this conversation.